testing/deploy-tenant-upgrade.sh

```diff
@@ -0,0 +1,145 @@
+lower_version="$1"
+upper_version="$2"
+namespace="tenant-lite"
+tenant="storage-lite"
+bucket="data"
+dummy="dummy.data"
+localport="9000"
+alias="minios3"
+
+# Port forward
+function port_forward() {
+  totalwait=0
+  echo 'Validating tenant pods are ready to serve'
+  for pod in `kubectl --namespace $namespace --selector=v1.min.io/tenant=$tenant get pod -o json |  jq '.items[] | select(.metadata.name|contains("'$tenant'"))| .metadata.name' | sed 's/"//g'`; do
+    while true; do
+      if kubectl --namespace $namespace logs pod/$pod | grep --quiet 'All MinIO sub-systems initialized successfully'; then
+        echo "$pod is ready to serve" && break
+      fi
+      sleep 5
+      totalwait=$((totalwait + 5))
+      if [ "$totalwait" -gt 305 ]; then
+        echo "Unable to validate pods after 5 minutes, exiting."
+        try false
+      fi
+    done
+  done
+
+  echo "Killing any current port-forward"
+  for pid in $(lsof -i :$localport | awk '{print $2}' | grep -o '\d*')
+  do
+    if [ -n "$pid" ] 
+    then
+      kill -9 $pid
+      echo "Killed previous port-forward process using port $localport: $pid"
+    fi
+  done
+
+  echo "Establishing port-forward"
+  kubectl port-forward service/$tenant-hl -n $namespace $localport:$localport &
+
+  echo 'start - wait for port-forward to be completed'
+  sleep 15
+  echo 'end - wait for port-forward to be completed'
+}
+
+# Preparing tenant for bucket manipulation
+function bootstrap_tenant() {
+  port_forward
+
+  # Obtain root credentials
+  TENANT_CONFIG_SECRET=$(kubectl -n $namespace get tenants $tenant -o jsonpath="{.spec.configuration.name}")
+  USER=$(kubectl -n $namespace get secrets "$TENANT_CONFIG_SECRET" -o go-template='{{index .data "config.env"|base64decode }}' | grep 'export MINIO_ROOT_USER="' | sed -e 's/export MINIO_ROOT_USER="//g' | sed -e 's/"//g')
+  PASSWORD=$(kubectl -n $namespace get secrets "$TENANT_CONFIG_SECRET" -o go-template='{{index .data "config.env"|base64decode }}' | grep 'export MINIO_ROOT_PASSWORD="' | sed -e 's/export MINIO_ROOT_PASSWORD="//g' | sed -e 's/"//g')
+
+  echo "Creating alias with user ${USER}"
+  mc alias set $alias https://localhost:$localport ${USER} ${PASSWORD} --insecure
+
+  echo "Creating bucket on tenant"
+  mc mb $alias/$bucket --insecure
+}
+
+# Upload dummy data to tenant bucket
+function upload_dummy_data() {
+  port_forward
+
+  echo "Uploading dummy data to tenant bucket"
+  cp ${SCRIPT_DIR}/deploy-tenant-upgrade.sh ${SCRIPT_DIR}/$dummy
+  mc cp ${SCRIPT_DIR}/$dummy $alias/$bucket/$dummy --insecure
+}
+
+# Download dummy data from tenant bucket
+function download_dummy_data() {
+  port_forward
+
+  echo "Download dummy data from tenant bucket"
+  mc cp $alias/$bucket/$dummy ${SCRIPT_DIR}/$dummy --insecure
+
+  if cmp "${SCRIPT_DIR}/deploy-tenant-upgrade.sh" "${SCRIPT_DIR}/$dummy"; then
+    echo "Operator upgrade test complete; no issue found"
+  else
+    echo "Operator upgrade test failed"
+    try false
+  fi
+}
+
+function main() {
```

Comment:
I think we should add an `echo` pointing what's going to happen, like `Testing upgrade of Opetor from $lower_version to $upper_version`

Reply:
Added as a separate function